ddtrace/tracer/span.go

```diff
@@ -692,6 +683,10 @@ func (s *Span) finish(finishTime int64) {
 		// already finished
 		return
 	}
+
+	s.serializeSpanLinksInMeta()
+	s.serializeSpanEvents()
```

Comment:
We also need to modify `AddSpanLink` and `AddEvent` to acquire the lock.

Reply:
True, let me fix that.

Reply:
Done in fe828cfa5c6b87e44b41bddfc67a1b427933a2cf.